src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -32,13 +61,149 @@ internal CrontabSchedule InnerSchedule
+            // transitions into and out of Daylight Saving Time.
+            var nowDateTime = now.LocalDateTime;
+            var nextDateTime = _cronSchedule.GetNextOccurrence(nowDateTime);
+
+            // This will apply the correct offset for the local time zone based on the DateTime.
+            var next = new DateTimeOffset(nextDateTime);
+
+            // Now we need to evaluate the possibility of Daylight Saving transitions.
+            // Example of DST transitions in 2018 for reference (from .NET source):
+            //
+            //         -=-=-=-=-=- Pacific Standard Time -=-=-=-=-=-=-
+            //   March 11, 2018                            November 4, 2018
+            // 2AM            3AM                        1AM              2AM
+            // |      +1 hr     |                        |       -1 hr      |
+            // | <invalid time> |                        | <ambiguous time> |
+            //                  [========== DST ========>)
+            if (TryAdjustInvalidTime(next, out DateTimeOffset? adjusted))
+            {
+                // "Spring Forward": "next" was invalid and was adjusted forward to a valid time.                
+                next = adjusted.Value;
+            }
+            else if (TryAdjustAmbiguousTime(now, next, out adjusted))
+            {
+                // "Fall Back": Either "now" or "next" was ambiguous and the time was adjusted.
+                next = adjusted.Value;
+            }
+
+            return next;
+        }
+
+        /// <summary>
+        /// If the provided DateTime is not valid, moves forward to the next occurrence in the schedule
+        /// until a valid DateTime is found. For example, 02:30 does not exist when "springing forward"
+        /// to Daylight Saving Time in the Pacific Standard Time Zone. Uses TimeZoneInfo.Local to make
+        /// the determination.
+        /// </summary>
+        /// <param name="next">The DateTimeOffset to evaluate.</param>
+        /// <param name="adjusted">The adjusted DateTimeOffset, if "next" is invalid.</param>
+        /// <returns></returns>
+        internal bool TryAdjustInvalidTime(DateTimeOffset next, out DateTimeOffset? adjusted)
+        {
+            adjusted = null;
+
+            if (TimeZoneInfo.Local.IsInvalidTime(next.DateTime))
+            {
+                // This forces the offset to be recalculated for the local time zone. It will also signal that we don't need to do the
+                // ambiguous time calculations as this cannot happen during ambiguous times.
+                adjusted = next.ToLocalTime();
+                _logger.LogDebug(new EventId(900, "DstInvalidTime"), $"The calculated next occurrence for the schedule '{_cronSchedule}' of '{next}' was determined to be an invalid time in the time zone '{TimeZoneInfo.Local.DisplayName}' due to Daylight Saving Time. The adjusted next occurrence is '{adjusted}'.");
```

Comment:
Would make sense to have these with `LoggerMessage.Define` (on a partial class or maybe using extension methods) and using structured logging? Not a hot path (well maybe for one hour a year) but I like having them listed with their EventId/EventName one after the other.

Reply:
Sure, I moved them to their own Logger class like others.